### PR TITLE
Change var popover position to follow input

### LINKED ISSUE #6397: Slice 2.2: Fix behavior variable popover position on long text fields. (Impact: can make entry impossible for long text fields)
https://www.notion.so/pixiebrix/Make-Variable-Popover-Generally-Available-e64deef53c894457901e88a606262f04?pvs=4#a5129cc65f0f494181a881ee29ed490a

### PR DESCRIPTION
**Pull request by turbochef**
## What does this PR do?

- Closes #6397
- Change the popover position so that it appears above or below the selected line.

## Demo

- [Loom](https://www.loom.com/share/eb4e2dea5d774e50998e6d0740fb1fb4?sid=be5621ea-097b-4d9a-881c-bb2126d884d1)

## Checklist

- [ ] Add tests
- [x] Designate a primary reviewer: @mnholtz

### CODE REVIEW

File: src/utils/textAreaUtils.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * @param element
+ * @param position
+ */
+export function getTextareaCaretCoordinates(
+  element: HTMLTextAreaElement,
+  position: number
+): Coordinates {
+  const div = document.createElement("div");
+  const span = document.createElement("span");
+  const computedStyle = getComputedStyle(element);
```
Comment (turbochef) [inline src/utils/textAreaUtils.ts:34]: I tried writing tests for the utilities but the computed dimensions don't seem to work in jest. Basically it always returns `{top: 0, left: 0}`

Comment (turbochef) [reply src/utils/textAreaUtils.ts:34]: Will try to compensate with rainforest

File: src/components/fields/schemaFields/widgets/varPopup/utils.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import { getTextareaCaretCoordinates } from "@/utils/textAreaUtils";
+
+/**
+ * Get a virtual element for the line that the caret is on.
+ * Essentially a box that has the same position and dimensions as the line that the caret is on.
+ * Used to position the floating UI menu.
+ */
+export function getSelectedLineVirtualElement(
+  textarea: HTMLTextAreaElement
+): VirtualElement {
```
Comment (mnholtz) [inline src/components/fields/schemaFields/widgets/varPopup/utils.ts:28]: It looks like this type `VirtualElement` is the only place this new `floating-ui` library is being used? Is it being used implicitly somewhere?

**Review by mnholtz — APPROVED** [review]

File: src/components/fields/schemaFields/widgets/varPopup/utils.ts (hunk 2 above)
Comment (turbochef) [reply src/components/fields/schemaFields/widgets/varPopup/utils.ts:28]: I'll remove `@floating-ui/react`. I installed it for a utility I didn't end up using.

**Pull request merged**